finality-aleph/src/new_network/connection_manager/discovery.rs

```diff
@@ -97,12 +97,19 @@ impl Discovery {
 
     /// Returns messages that should be sent as part of authority discovery at this moment.
     pub fn discover_authorities(&mut self, handler: &SessionHandler) -> Vec<DiscoveryCommand> {
+        let authentication = handler.authentication();
+
+        if authentication == None {
+            return Vec::new();
```

Comment:
@timorl Are you fine with changing this in other small PR? Would prefer to merge this one without targeting this case in better way. I would prepare other till Tuesday

Reply:
Nah, this ain't that hard:
```
let authentication = match handler.authentication() {
 Some(authentication) => authentication,
 None => return Vec::new(),
}

Reply:
Same below.

Reply:
Your code aren't doing anything else than mine XD

Reply:
But it's massively sexier.